java/core/src/main/java/com/lancedb/lance/Dataset.java

```diff
@@ -771,14 +771,9 @@ public Map<String, String> getConfig() {
    * @param tableConfig the config to update
    */
   public void updateConfig(Map<String, String> tableConfig) {
-    try (LockManager.WriteLock writeLock = lockManager.acquireWriteLock()) {
-      Preconditions.checkArgument(nativeDatasetHandle != 0, "Dataset is closed");
-      nativeUpdateConfig(tableConfig);
-    }
+    updateConfig(tableConfig, false);
```

Comment:
Instead of adding a ton of new JNI methods for setting all these update maps, we can just mark `updateConfig` as `@Deprecated` and route it to call the new transaction builder instead, something like

```java
dataset
  .newTransactionBuilder()
  .operation(UpdateConfig.builder()
    .configUpdates(...)
    .build()
  )
  .commit()
``` 

and let users use the transaction builder API for performing other types of metadata updates.